packages/tools/viewer/src/viewer.ts

```diff
@@ -1073,6 +1091,27 @@ export class Viewer implements IDisposable {
                 resetWorldBounds: () => {
                     cachedWorldBounds.length = 0;
                 },
+                get selectedAnimation() {
+                    return selectedAnimation;
+                },
+                selectAnimation: (index: number) => {
+                    const startAnimation = this.isAnimationPlaying;
+                    if (this._activeAnimation) {
+                        this._activeAnimation.pause();
+                        this._activeAnimation.goToFrame(0);
+                    }
+
+                    selectedAnimation = index;
+
+                    if (this._activeAnimation) {
+                        this._activeAnimation.goToFrame(0);
+                        this._activeAnimation.play(true);
+
+                        if (!startAnimation) {
+                            this.pauseAnimation();
+                        }
+                    }

```

Comment:
As is this seems to mix Model scoped state with Viewer scoped state. I think what I was imagining was just having:
```ts
let activeAnimation = assetContainer.animationGroups[selectedAnimation];
const startAnimation = activeAnimation?.isPlaying ?? false;
if (activeAnimation) {
  activeAnimation.pause();
  activeAnimation.goToFrame(0);
}

selectedAnimation = index;
activeAnimation = assetContainer.animationGroups[selectedAnimation];

if (activeAnimation) {
  activeAnimation.goToFrame(0);
  activeAnimation.play(true);

  if (!startAnimation) {
    activeAnimation.pause();
  }
}
```